src/pyfmi/fmi3.pyx

```diff
@@ -465,6 +532,78 @@ cdef class FMUModelBase3(FMI_BASE.ModelBase):
             Calls the low-level FMI function: fmi3SetTime or fmi3GetTime.
     """)
 
+    def get_variable_declared_type(self, variable_name):
+        """
+        Return the given variables declared type.
+
+        Parameters::
+
+            variable_name --
+                The name of the variable.
+
+        Returns::
+
+            The declared type.
+        """
+        cdef FMIL3.fmi3_import_variable_t* variable
+        cdef FMIL3.fmi3_value_reference_t  vr
+        cdef FMIL3.fmi3_import_variable_typedef_t* variable_type
+        cdef FMIL3.fmi3_base_type_enu_t    basetype
+        cdef FMIL3.fmi3_import_enumeration_typedef_t * enumeration_type
+        cdef FMIL3.fmi3_import_integer_typedef_t * integer_type
```

Comment:
A couple of these are unused for now, I propose to have a pass over these and remove any not (yet) used.